firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -606,6 +606,8 @@ public void notifyLocalViewChanges(List<LocalViewChanges> viewChanges) {
               TargetData updatedTargetData =
                   targetData.withLastLimboFreeSnapshotVersion(lastLimboFreeSnapshotVersion);
               queryDataByTarget.put(targetId, updatedTargetData);
+
+              targetCache.updateTargetData(updatedTargetData);
```

Comment:
Only if you feel like it, could you try to call `shouldPersistTargetData(targetData, updatedTargetData, ..)` here too. And then add similar logic handling version update, but apply to lastLimboFreeVersion, in `shouldPersistTargetData`?

This way we only persist target data when `lastLimboFreeVersion` advances over a big enough interval, otherwise it is persisted pretty regularly, defeating the purpose of having this optimization in the first place.

Please let me know if you don't want to bother with this, I can take it from here.

Reply:
Yeah, I was thinking along the same lines. My original reason to not do that was that shouldPersistTargetData takes in changes of type firestore/remote/TargetChange and the notifyLocalViewChanges has changes of type firestore/local/LocalViewChanges

I can make a new function, but also wasn't sure if we care about the other checks shouldPersistTargetData does, like: resume token, snapshot version times, and changes? Or are all of those ok to still be handled by the original shouldPersistTargetData inside of applyRemoteEvent?

ie Do we only need to do a check on lastLimboFreeSnapshotVersion delta within notifyLocalViewChanges?

Reply:
@wu-hui any preference or suggestion?

Reply:
Yes, we only need to check for lastLimboFreeSnapshotVersion within notifyLocalViewChanges and leave everything else unchanged.

You could do this in shouldPersisteTargetData, by adding an extra check for lastLimboFreeSnapshotVersion in that method. And call it from notifyLocalViewChanges.

Alternatively, you could explicitly check for lastLimboFreeSnapshotVersion from within notifyLocalViewChanges, but I think keeping the logic in one place makes slightly more sense here.